### ISSUE TITLE
Expo config plugin overrides AppDelegate imports from other plugins

### ISSUE THREAD

**Issue by gunnartorfis** [zoontek/react-native-bootsplash#580]
### Before submitting a new issue

- [X] I tested using the latest version of the library, as the bug might be already fixed.
- [X] I tested using a [supported version](https://github.com/reactwg/react-native-releases/blob/main/docs/support.md) of react native.
- [X] I checked for possible duplicate issues, with possible answers.

### Bug summary

When adding the config plugin as the React Conf app does, this library's imports are overriding other plugins. 
If I move the other plugin (in this case the adyen plugin as provided in the repro repo) below the bootsplash one, it works fine. I'm not sure if it is this config plugin's issue or the adyen one. Feel free to close the issue if it is the adyen config plugin's fault. 

**Before adding this library's config plugin**
<img width="414" alt="Screenshot 2024-05-19 at 16 04 18" src="https://github.com/zoontek/react-native-bootsplash/assets/5333875/33e3c6ed-8f40-4572-9938-48cd6195bdf6">

**After adding the config plugin**
<img width="550" alt="Screenshot 2024-05-19 at 16 05 41" src="https://github.com/zoontek/react-native-bootsplash/assets/5333875/a879757e-a527-411e-bb73-d776bd222d2c">

### Library version

6.0.0-beta.8

### Environment info

```shell
System:
  OS: macOS 14.3.1
  CPU: (10) arm64 Apple M1 Pro
  Memory: 778.02 MB / 32.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.11.0
    path: ~/.nvm/versions/node/v20.11.0/bin/node
  Yarn:
    version: 4.0.0
    path: ~/.nvm/versions/node/v20.11.0/bin/yarn
  npm:
    version: 10.2.4
    path: ~/.nvm/versions/node/v20.11.0/bin/npm
  Watchman: Not Found
Managers:
  CocoaPods:
    version: 1.14.2
    path: /Users/gunnartorfi/.rvm/gems/ruby-2.7.6/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.5
      - iOS 17.5
      - macOS 14.5
      - tvOS 17.5
      - visionOS 1.2
      - watchOS 10.5
  Android SDK: Not Found
IDEs:
  Android Studio: 2023.1 AI-231.9392.1.2311.11330709
  Xcode:
    version: 15.4/15F31d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 21.0.1
    path: /opt/homebrew/opt/openjdk/bin/javac
  Ruby:
    version: 2.7.6
    path: /Users/gunnartorfi/.rvm/rubies/ruby-2.7.6/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.74.1
    wanted: 0.74.1
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false
```

### Steps to reproduce

1. Install the library
2. Add the config plugin 
3. Prebuild (CNG)

### Reproducible sample code

```js
https://github.com/gunnartorfis/bootsplash-expo-repro
```

**Comment by zoontek** [comment]
@gunnartorfis Can you update the second screenshot link?

**Comment by gunnartorfis** [comment]
@zoontek sorry, updated!

**Comment by zoontek** [comment]
@gunnartorfis Which versions of `expo` / `@adyen/react-native` are you using? I cannot reproduce it with my test app using Expo 50. I personally use [`expo` `mergeContents` helper](https://github.com/zoontek/react-native-bootsplash/blob/06f995407eaacfd3b956f12d36c97f47024ff275/src/generate.ts#L1428), Adyen uses [RegExps](https://github.com/Adyen/adyen-react-native/blob/develop/plugin/src/setImport.ts). Not sure which one conflict with the other here.

Could you also indicate which version of `@expo/config-plugins` package is resolved in your package manager lock file?

<img width="1129" alt="Screenshot 2024-05-19 at 19 53 36" src="https://github.com/zoontek/react-native-bootsplash/assets/1902323/b9a540a1-cdc8-4840-99ce-79da0ccfe30e">

**With the adyen plugin then the bootsplash one:**
<img width="1008" alt="Screenshot 2024-05-19 at 19 45 10" src="https://github.com/zoontek/react-native-bootsplash/assets/1902323/5ae685ed-7986-4999-af18-aaa4d8696b30">

**With the bootsplash plugin then the adyen one:**
<img width="1010" alt="Screenshot 2024-05-19 at 19 46 13" src="https://github.com/zoontek/react-native-bootsplash/assets/1902323/b95bad95-c769-41a8-b93f-aaac37ead87d">

**Comment by gunnartorfis** [comment]
@zoontek I'm using Expo SDK 51, config plugins 8.0.4. 

I wasn't aware that there was a mergeContents helper, neat! 
This isn't a huge issue since the workaround is to change the order. If you give the green light, I'll open up a similar issue on the Adyen repo.

**Comment by zoontek** [comment]
I updated to Expo 51 and tried it again. Still the same on my side.

**With the adyen plugin then the bootsplash one:**
<img width="1010" alt="Screenshot 2024-05-20 at 09 55 45" src="https://github.com/zoontek/react-native-bootsplash/assets/1902323/d07a565b-41bd-4a41-b579-ed329274623d">

**With the bootsplash plugin then the adyen one:**
<img width="1010" alt="Screenshot 2024-05-20 at 09 56 42" src="https://github.com/zoontek/react-native-bootsplash/assets/1902323/c663e1ea-6c1c-4ca1-98d6-8d7a99425ddd">

I used `react-native-bootsplash@6.0.0-beta.8` and `@adyen/react-native@2.0.0-rc.1`

**Comment by zoontek** [comment]
@gunnartorfis Could you open an identical issue on the Adyen package repository?

**Comment by gunnartorfis** [comment]
@zoontek did you use the config plugin from the Adyen repo or the one from my repro repo? 

I just noticed that Adyen now has an built-in one. I took mine from the PR that added the config plugin, and since it took a while to merge I copied it as it was at the time. I just switched to the built-in one from the Adyen library, and it works! 

I'm sorry for the trouble, I should've realized this sooner 🫠
